browser/ui/toolbar/brave_app_menu_model.cc

```diff
@@ -284,18 +250,191 @@ void BraveAppMenuModel::InsertBraveMenuItems() {
+  }
+
+  if (IsCommandIdEnabled(IDC_MANAGE_EXTENSIONS)) {
+    InsertItemWithStringIdAt(
+        GetIndexOfCommandId(IDC_SHOW_DOWNLOADS).value() + 1,
+        IDC_MANAGE_EXTENSIONS, IDS_SHOW_EXTENSIONS);
+  }
+}
+
+void BraveAppMenuModel::BuildMoreToolsSubMenu() {
+  ui::SimpleMenuModel* more_tools_menu_model =
+      static_cast<ui::SimpleMenuModel*>(
+          GetSubmenuModelAt(GetIndexOfCommandId(IDC_MORE_TOOLS_MENU).value()));
+  DCHECK(more_tools_menu_model);
+
+  size_t next_target_index = 0;
+  bool need_separator = false;
+
+  // Create New Profile
+  if (IsCommandIdEnabled(IDC_ADD_NEW_PROFILE)) {
+    more_tools_menu_model->InsertItemWithStringIdAt(
+        next_target_index++, IDC_ADD_NEW_PROFILE, IDS_ADD_NEW_PROFILE);
+    need_separator = true;
+  }
+
+  // Open Guest Window
+  if (IsCommandIdEnabled(IDC_OPEN_GUEST_PROFILE)) {
+    more_tools_menu_model->InsertItemWithStringIdAt(
+        next_target_index++, IDC_OPEN_GUEST_PROFILE, IDS_OPEN_GUEST_PROFILE);
+    need_separator = true;
+  }
+
+  if (need_separator) {
+    more_tools_menu_model->InsertSeparatorAt(next_target_index++,
+                                             ui::NORMAL_SEPARATOR);
+    need_separator = false;
+  }
+
+#if defined(TOOLKIT_VIEWS)
+  if (sidebar::CanUseSidebar(browser())) {
+    sub_menus_.push_back(std::make_unique<SidebarMenuModel>(browser()));
+    more_tools_menu_model->InsertSubMenuWithStringIdAt(
+        next_target_index++, IDC_SIDEBAR_SHOW_OPTION_MENU,
+        IDS_SIDEBAR_SHOW_OPTION_TITLE, sub_menus_.back().get());
+    more_tools_menu_model->InsertSeparatorAt(next_target_index++,
+                                             ui::NORMAL_SEPARATOR);
+  }
+#endif
+
+  if (media_router::MediaRouterEnabled(browser()->profile())) {
+    more_tools_menu_model->InsertItemWithStringIdAt(
+        next_target_index++, IDC_ROUTE_MEDIA, IDS_MEDIA_ROUTER_MENU_ITEM_TITLE);
+    need_separator = true;
+  }
+
+  // Insert sync menu
+  if (IsCommandIdEnabled(IDC_SHOW_BRAVE_SYNC)) {
+    more_tools_menu_model->InsertItemWithStringIdAt(
+        next_target_index++, IDC_SHOW_BRAVE_SYNC, IDS_SHOW_BRAVE_SYNC);
+    need_separator = true;
+  }
+
+  if (need_separator) {
+    more_tools_menu_model->InsertSeparatorAt(next_target_index++,
+                                             ui::NORMAL_SEPARATOR);
+    need_separator = false;
+  }
+
+  if (auto index =
+          more_tools_menu_model->GetIndexOfCommandId(IDC_TASK_MANAGER)) {
+    more_tools_menu_model->InsertItemWithStringIdAt(*index, IDC_DEV_TOOLS,
+                                                    IDS_DEV_TOOLS);
+  }
+}
+
+void BraveAppMenuModel::BuildHelpSubMenu() {
+  // Put help sub menu above the settings menu.
+  if (const auto index = GetIndexOfCommandId(IDC_OPTIONS)) {
+    sub_menus_.push_back(std::make_unique<BraveHelpMenuModel>(this));
+    InsertSubMenuWithStringIdAt(*index, IDC_HELP_MENU, IDS_HELP_MENU,
+                                sub_menus_.back().get());
+  }
```

Comment:
`else { NOTREACHED(); }` or just use `GetIndexOfCommandId(IDC_OPTIONS)` without checks, otherwise we may miss HELP menu and no one will notice.

Reply:
Test code could detect if there is no help menu - https://github.com/brave/brave-core/pull/19833/files#diff-ea1e30fc1cb59aa92e1e9fbc0ad4ede54d1bb53942338dde4a0486ae60b4be18R215